src/contrib/persistence/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -0,0 +1,421 @@
+                            await MaxMarkedForDeletionMaxPersistenceIdQuery(db,
+                                persistenceId).FirstOrDefaultAsync();
+                        if (_journalConfig.DaoConfig.SqlCommonCompatibilityMode)
+                        {
+                            await db.GetTable<JournalMetaData>()
+                                .InsertOrUpdateAsync(() => new JournalMetaData()
+                                    {
+                                        PersistenceId = persistenceId,
+                                        SequenceNumber =
+                                            maxMarkedDeletion
+                                    },
+                                    jmd => new JournalMetaData()
+                                    {
+                                        PersistenceId = persistenceId,
+                                        SequenceNumber = maxMarkedDeletion
+                                    },
+                                    () => new JournalMetaData()
+                                    {
+                                        PersistenceId = persistenceId,
+                                        SequenceNumber = maxMarkedDeletion
+                                    });
+                        }
+
+                        if (logicalDelete == false)
+                        {
+                            await db.GetTable<JournalRow>()
+                                .Where(r =>
+                                    r.persistenceId == persistenceId &&
+                                    (r.sequenceNumber <= maxSequenceNr &&
+                                     r.sequenceNumber <
+                                     maxMarkedDeletion
+                                         )).DeleteAsync();
+                        }
+
+                        if (_journalConfig.DaoConfig.SqlCommonCompatibilityMode)
+                        {
+                            await db.GetTable<JournalMetaData>()
+                                .Where(r =>
+                                    r.PersistenceId == persistenceId &&
+                                    r.SequenceNumber <
+                                    maxMarkedDeletion)
+                                .DeleteAsync();
+                        }
+
+                        await transaction.CommitAsync();
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.Error(ex,"Error on delete!");
+                        try
+                        {
+                            await transaction.RollbackAsync();
+                        }
+                        catch (Exception )
+                        {
+                            //If rollback fails, Don't throw as it will mask
+                            //original exception
+                        }
+                        
+                        throw;
+                    }
```

Comment:
Should we really be async here? and/or make async optional?

Concern is that (with SQLServer, SQLite, anything else with lots of locking) we don't know how quickly we will get back to this task chain under load, leaving the table(s) locked even longer.

I can say from experience running the driver live, this is a thing; turning off delete compatibility (delete compatibility increases the db ops here) solved a bunch of stability issues on a project I've been working on.